integration-tests/cucumber-tests-protocol-iec60870/src/test/java/org/opensmartgridplatform/cucumber/protocol/iec60870/glue/steps/ProcessImageSteps.java

```diff
@@ -0,0 +1,100 @@
+import org.opensmartgridplatform.cucumber.protocol.iec60870.domain.Iec60870ProcessImage;
+import org.opensmartgridplatform.cucumber.protocol.iec60870.mock.Iec60870MockServer;
+import org.opensmartgridplatform.iec60870.Iec60870InformationObjectType;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class ProcessImageSteps {
+
+  @Autowired private Iec60870MockServer mockServer;
+
+  @DataTableType
+  public Iec60870ProcessImage processImageTable(final DataTable table) {
+    final Map<Integer, Iec60870InformationObject> informationObjects =
+        table.asMaps().stream()
+            .map(this::informationObjectEntry)
+            .collect(Collectors.toMap(Iec60870InformationObject::getAddress, Function.identity()));
+    return new Iec60870ProcessImage(informationObjects);
+  }
+
+  @DataTableType
+  public Iec60870InformationObject informationObjectEntry(final Map<String, String> entry) {
+    return new Iec60870InformationObject(
+        informationObjectAddress(entry),
+        informationObjectType(entry),
+        informationElementValue(entry));
+  }
+
+  @Given("a process image on the IEC60870 server")
+  public void givenAProcessImageOnTheIec60870Server(final Iec60870ProcessImage processImage) {
+
+    final Consumer<Iec60870InformationObject> setInformationObject =
+        informationObject ->
+            this.mockServer
+                .getRtuSimulator()
+                .initInformationObject(
+                    informationObject.getAddress(),
+                    informationObject.getType(),
+                    informationObject.getValue());
+
+    processImage.getInformationObjects().values().forEach(setInformationObject);
+  }
+
+  @When("the process image on the IEC60870 server changes")
+  public void whenTheProcessImageOnTheIec60870ServerChanges(
+      final Iec60870InformationObject informationObject) {
+
+    this.mockServer
+        .getRtuSimulator()
+        .updateInformationObject(
+            informationObject.getAddress(),
+            informationObject.getType(),
+            informationObject.getValue());
+  }
+
+  private static int informationObjectAddress(final Map<String, String> entry) {
+    return ReadSettingsHelper.getInteger(entry, ProtocolIec60870Keys.INFORMATION_OBJECT_ADDRESS);
+  }
+
+  private static Iec60870InformationObjectType informationObjectType(
+      final Map<String, String> entry) {
+
+    return ReadSettingsHelper.getEnum(
+        entry, ProtocolIec60870Keys.INFORMATION_OBJECT_TYPE, Iec60870InformationObjectType.class);
+  }
+
+  private static Object informationElementValue(final Map<String, String> entry) {
+    return informationElementValue(entry.get(ProtocolIec60870Keys.INFORMATION_ELEMENT_VALUE));
+  }
+
+  private static Object informationElementValue(final String value) {
+    final String profile = (String) ScenarioContext.current().get(PROFILE);
+    if ("light_measurement_device".equals(profile)) {
```

Comment:
Assuming this is the same profile as specified in the scenario hooks, it might be a good idea to use a constant for this profile identifier.

This way of dealing with the values does not look really future proof, or will all profiles always have values of the same type for all information objects? Might be good enough for now though, as I don't think it would be a trivial fix, but we may need to keep this in mind with future changes.

Reply:
Created Iec60870ProfileType enum.